src/hotspot/share/gc/serial/markSweep.hpp

```diff
@@ -199,6 +198,8 @@ class PreservedMark {
   markWord _mark;
 
 public:
+  PreservedMark(oop obj, markWord mark) : _obj(obj), _mark(mark) {}
+
   void init(oop obj, markWord mark) {
```

Comment:
Do we even need this `init` function anymore?  Even if such an assignment is still needed, it could just use assignment from a newly constructed object, e.g. `pm = PreservedMark(obj, mark);`.  The only place I found where it is used is preserve_marks, which can use the assignment form instead.